tm2/pkg/telemetry/init.go

```diff
@@ -5,41 +5,64 @@ package telemetry
 
 import (
 	"fmt"
-	"sync/atomic"
+	"log/slog"
 
 	"github.com/gnolang/gno/tm2/pkg/telemetry/config"
 	"github.com/gnolang/gno/tm2/pkg/telemetry/metrics"
-)
-
-var (
-	globalConfig         config.Config
-	telemetryInitialized atomic.Bool
+	"github.com/gnolang/gno/tm2/pkg/telemetry/traces"
+	sdkMetric "go.opentelemetry.io/otel/sdk/metric"
+	sdkTrace "go.opentelemetry.io/otel/sdk/trace"
 )
 
 // MetricsEnabled returns true if metrics have been initialized
 func MetricsEnabled() bool {
-	return globalConfig.MetricsEnabled
+	return config.GetGlobalConfig().MetricsEnabled
+}
+
+// TracesEnabled returns true if traces have been initialized
+func TracesEnabled() bool {
+	return config.GetGlobalConfig().TracesEnabled
 }
 
 // Init initializes the global telemetry
-func Init(c config.Config) error {
-	// Check if the metrics are enabled at all
-	if !c.MetricsEnabled {
-		return nil
+func Init(c config.Config, logger *slog.Logger) (*sdkTrace.TracerProvider, *sdkMetric.MeterProvider, error) {
```

Comment:
Passing a logger just to log that something happened in the func is overkill

Reply:
This function should also be split up into 2 different ones, one for the tracing, and one for the metrics -- because in the current state you still require the caller to check whether something is `nil`